servant-server/src/Servant/Server/Internal.hs

```diff
@@ -134,8 +135,8 @@ processMethodRouter :: forall a. ConvertibleStrings a B.ByteString
                     -> Maybe [(HeaderName, B.ByteString)]
                     -> Request -> RouteResult Response
 processMethodRouter handleA status method headers request = case handleA of
-  Nothing -> failWith UnsupportedMediaType
-  Just (contentT, body) -> succeedWith $ responseLBS status hdrs bdy
+  Nothing -> FailFatal err406
```

Comment:
@kosmikus They would be equivalent only if in

``` haskell
     Put '[Foo] T
:<|> Put '[Bar] T
```

both endpoints do the thing that the "one endpoint only" version does. Right?